terminology/src/main/java/au/csiro/pathling/caching/InfinispanStorage.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 Commonwealth Scientific and Industrial Research
+ * Organisation (CSIRO) ABN 41 687 119 230.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package au.csiro.pathling.caching;
+
+import java.io.Closeable;
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import org.apache.http.client.cache.HttpCacheEntry;
+import org.apache.http.client.cache.HttpCacheStorage;
+import org.apache.http.client.cache.HttpCacheUpdateCallback;
+import org.infinispan.Cache;
+import org.infinispan.manager.DefaultCacheManager;
+
+/**
+ * An abstract {@link HttpCacheStorage} implementation that uses embedded Infinispan as the
+ * underlying storage. Subclasses need to initialize the cache and cache manager.
+ *
+ * @author John Grimes
+ */
+public abstract class InfinispanStorage implements HttpCacheStorage, Closeable {
+
```

Comment:
Since the only aspect the subclasses customise it the construction maybe we do not need them. 
We could just have this class with members private and final, a private constructor and
two static methods that construct memory and disk instances.

Reply:
I decided not to make this change, as it makes it harder to test.